src/python_testing/TC_TIMESYNC_2_1.py

```diff
@@ -34,40 +34,51 @@ def pics_TC_TIMESYNC_2_1(self) -> list[str]:
 
     @async_test_body
     async def test_TC_TIMESYNC_2_1(self):
+        endpoint = 0
 
-        endpoint = self.user_params.get("endpoint", 0)
+        features = await self.read_single_attribute(dev_ctrl=self.default_controller, node_id=self.dut_node_id,
+                                                    endpoint=endpoint, attribute=Clusters.TimeSynchronization.Attributes.FeatureMap)
+
+        self.supports_time_zone = bool(features & Clusters.TimeSynchronization.Bitmaps.Feature.kTimeZone)
+        self.supports_ntpc = bool(features & Clusters.TimeSynchronization.Bitmaps.Feature.kNTPClient)
+        self.supports_ntps = bool(features & Clusters.TimeSynchronization.Bitmaps.Feature.kNTPServer)
+        self.supports_trusted_time_source = bool(features & Clusters.TimeSynchronization.Bitmaps.Feature.kTimeSyncClient)
+        timesource_attr_id = Clusters.Objects.TimeSynchronization.Attributes.TimeSource.attribute_id
+
+        # Reading attributes from DUT
+        timesync_attr_list = Clusters.Objects.TimeSynchronization.Attributes.AttributeList
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [timesync_attr_list])
```

Comment:
There's a function in matter_testing_support called "read_single_attribute_check_success" (or...something like that), which is what you'll want to use here. It will extract the attribute info from the endpoint

Reply:
Thank you, updated to using read_single_attribute_check_success() from matter_testing_support works very nicely it appears and made this section of the code much cleaner now.